### PR TITLE
#162948909 Email verification

### PR DESCRIPTION
**Pull request by tesh254**
#### What does this PR do?

This PR verifies a  user after signup. It creates a verification link which is sent to the user upon registration then the user

#### Description of Task to be completed?

* The users can verify their account after registration.
* A link is sent to the email. After the user clicks the link they will be able to activate their account.
* However users who havent received the verification link will not be able to log in as the account will not be verified.
#### How should this be manually tested?

* Clone ah-alpha repo $ git clone https://github.com/andela/ah-alpha.git
* Run `pip install -r requirements.txt` to update requirements.txt 
* git checkout ft-email-verify-162948909
* set up postgres database on the psql command line
* Create a virtual environment and create a .env file and add th database credentials
* export db credentials. please note that there is no spaces before and after =
``` export DB_NAME="name of your database"
      export DB_PASSWORD="passord to your database user"
      export DB_HOST="your local server eg (127.0.0.1)"
      export DB_USER="your database username eg (postgres)"
```
* Note- if you have no password set for your database user leave the password field blank example: 
  `export DB_PASSWORD=""`

On the project's root directory run `python manage.py runserver`
send a POST request to the server http://127.0.0.1:8000/api/users/ whith the following in the body.
```
{
    "user":{
        "username":"Username",
        "email":"example@email.com",
        "password":"Password"
    }
}
```
* Open a mail to check for receipt of email verification

#### What are the relevant pivotal tracker stories?
[#162948909](https://www.pivotaltracker.com/story/show/162948909)

### CODE REVIEW

**Review by danuluma — APPROVED** [review]

File: authors/apps/authentication/models.py [hunk 1]
```diff
@@ -74,6 +80,7 @@ class User(AbstractBaseUser, PermissionsMixin):
     # log into the Django admin site. For most users, this flag will always be
     # falsed.
     is_staff = models.BooleanField(default=False)
+    is_confirmed = models.BooleanField(default=False)
```
Comment (Ruiru11) [inline authors/apps/authentication/models.py]: add a comment for this so that everyone understands why its there

**Review by Ruiru11 — APPROVED** [review]

File: authors/apps/authentication/models.py (hunk 1 above)
Comment (tesh254) [reply authors/apps/authentication/models.py]: Noted

**Review by boltonotieno — APPROVED** [review]

**Review by jamesbeamie — APPROVED** [review]

File: authors/apps/authentication/backends.py [hunk 2]
```diff
@@ -1,10 +1,31 @@
-# import jwt
+import jwt
 #
```
Comment (frankip) [inline authors/apps/authentication/backends.py]: @tesh254 remove commented out lines of code

File: authors/apps/authentication/models.py [hunk 3]
```diff
@@ -4,51 +4,57 @@
 
 from django.conf import settings
 from django.contrib.auth.models import (
-    AbstractBaseUser, BaseUserManager, PermissionsMixin
+        AbstractBaseUser, BaseUserManager, PermissionsMixin
 )
 from django.db import models
+# from rest_framework_jwt.settings import api_settings
```
Comment (frankip) [inline authors/apps/authentication/models.py]: Also the same here

File: authors/apps/authentication/serializers.py [hunk 4]
```diff
@@ -23,78 +26,125 @@ class Meta:
...
+
+		# Django provides a flag on our `User` model called `is_active`. The
+		# purpose of this flag to tell us whether the user has been banned
+		# or otherwise deactivated. This will almost never be the case, but
+		# it is worth checking for. Raise an exception in this case.
+		if not user.is_active:
+			raise serializers.ValidationError(
+				'This user has been deactivated.'
+			)
+		# if not user.is_confirmed:
```
Comment (frankip) [inline authors/apps/authentication/serializers.py]: Also the same goes for this part

**Review by frankip — COMMENTED** [review]
@tesh254 @Deekerubo Good work on the test and the implementation however you need to remove the commented out ode and add the .env variables to .env_sample file. And also fix the conflicts.

File: authors/apps/authentication/backends.py (hunk 2 above)
Comment (tesh254) [reply authors/apps/authentication/backends.py]: I have removed the commented lines

**Review by jamesbeamie — APPROVED** [review]

**Pull request merged**